trashies.go

```diff
@@ -0,0 +1,85 @@
+package main
+
+import (
+	"net/http"
+	"log"
+	"github.com/gorilla/mux"
+	"encoding/json"
+	"html/template"
+)
+
+type Person struct {
+	ID        string   `json:"id,omitempty"`
+	FirstName string   `json:"firsname,omitempty"`
+	LastName  string   `json:"lastname,omitempty"`
+	Address   *Address `json:"address,omitempty"`
+}
+
+type Address struct {
+	City  string `json:"city,omitempty"`
+	State string `json:"state,omitempty"`
+}
+
+var people []Person
+
+func GetPeople(w http.ResponseWriter, r *http.Request) {
+	json.NewEncoder(w).Encode(people)
+}
+func GetPerson(w http.ResponseWriter, r *http.Request) {
+	params := mux.Vars(r)
+	for _, item := range people {
+		if item.ID == params["id"] {
+			json.NewEncoder(w).Encode(item)
+			return
+		}
+	}
+	json.NewEncoder(w).Encode(&Person{})
+}
+
+func CreatePerson(w http.ResponseWriter, r *http.Request) {
+	params := mux.Vars(r)
+	var person Person
+	_ = json.NewDecoder(r.Body).Decode(&person)
+	person.ID = params["id"]
+	people = append(people, person)
+	json.NewEncoder(w).Encode(people)
+}
+
+func DeletePerson(w http.ResponseWriter, r *http.Request) {
+	params := mux.Vars(r)
+	for index, item := range people {
+		if item.ID == params["id"] {
+			people = append(people[:index], people[index+1:]...)
+			break
+		}
+		json.NewEncoder(w).Encode(people)
+	}
+}
+
+func launch(w http.ResponseWriter, r *http.Request) {
+	t, _ := template.ParseFiles("index.html")
+	t.Execute(w, nil)
+	log.Println("YEEE")
+}
+
+func main() {
+
+	cssHandler := http.FileServer(http.Dir("./css/"))
+	jsHandler := http.FileServer(http.Dir("./js/"))
+
+	http.Handle("/css/", http.StripPrefix("/css/", cssHandler))
+	http.Handle("/js/", http.StripPrefix("/js/", jsHandler))
+
+	people = append(people, Person{ID: "1", FirstName: "John", LastName: "Doe", Address: &Address{City: "City X", State: "State X"}})
```

Comment:
Most of this will need to get deleted but it also does a good job of explaining how to interact with Go so I temporarily left it in. Let me know if you want me to pull it out.